models/models.go

```diff
@@ -0,0 +1,26 @@
+package models
+
+// AXFRResourceRecordSet is a representation of record name -> string
+type AXFRResourceRecordSet map[string][]string
+
+// Even though it would be nice to have the model map to something better than a string
+// (a la an actual SRV, or A struct).
+// This is the internal structure of how mesos-dns works today and the transformation of string -> DNS Struct
+// happens on actual query time. Why this logic happens at query time? Who knows.
+
+// AXFRRecords are the As, and SRVs that actually make up the Mesos-DNS zone
+type AXFRRecords struct {
+	As   AXFRResourceRecordSet
+	SRVs AXFRResourceRecordSet
+}
+
+// AXFR is a rough representation of a "transfer" of the Mesos-DNS data
+type AXFR struct {
+	TTL            uint32
+	Serial         uint32
+	RefreshSeconds int
```

Comment:
RefreshSeconds is a hint of refreshseconds from the config. When someone polls this endpoint, it hints to them to never poll it more often than refreshseconds. TTL tells someone that when they generate the zone, this is the TTL they should put on the records.

TTL should probably be something like 5 seconds if refreshseconds is ~60s or less, to get a decent interval for refreshing.